datasetviewer/dimension/DimensionWidget.py

```diff
@@ -0,0 +1,117 @@
+from datasetviewer.dimension.interfaces.DimensionViewInterface import DimensionViewInterface
+from datasetviewer.dimension.DimensionPresenter import DimensionPresenter
+from datasetviewer.dimension.Command import Command
+from PyQt5.QtWidgets import QLabel, QPushButton, QSlider, QSpinBox, QWidget, QHBoxLayout, QSpacerItem, QSizePolicy
+from PyQt5.QtCore import Qt
+
+class DimensionWidget(QWidget, DimensionViewInterface):
+
+    def __init__(self, dim_name, dim_size, parent):
+
+        QWidget.__init__(self, parent)
+
+        # Create a DimensionPresenter and give it a reference to the DimensionView
+        self._presenter = DimensionPresenter(self, dim_name)
+
+        self.dim_name = dim_name
+        self.dim_size = dim_size
+
+        layout = QHBoxLayout()
```

Comment:
Since we are using the Qt object model to programatically construct the UI, it might be useful to comment what each of these Qt objects is doing in terms of display. I'm ok reading it as i'm very used to Qt, but other developers might find it harder.